core/include/traccc/finding/details/debug_output.hpp

```diff
@@ -0,0 +1,28 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2023-2025 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+#include "traccc/utils/logging.hpp"
+
+namespace traccc {
+enum class propagate_to_next_surface_exit_mode {
+    HOLE_LIMIT_REACHED,
+    BRANCH_LIMIT_REACHED,
+    MAX
+};
+
+struct propagate_to_next_surface_debug {
+    void reset();
+
+    unsigned int m_failure_count[static_cast<std::size_t>(
+        propagate_to_next_surface_exit_mode::MAX)];
+};
```

Comment:
This is fairly subjective, but I'd prefer to do this:

```c++
namespace traccc::details {
struct propagate_to_next_surface_debug_info {
    enum class exit_mode {...};
    unsigned int m_failure_count...;
    void reset();
};
}
```

While talking about this... I assume `reset()` is not meant to be called in device code. But if we explicitly want to prevent users from doing it, it might be helpful to use `TRACCC_HOST` on it. If not, if technically it may be usable in device code in some circumstances, I would put `TRACCC_HOST_DEVICE` on it right away. 🤔